### ISSUE TITLE
Labels missing from example training dataset

### ISSUE THREAD

**Issue by rachel3834** [Professor-G/MicroLIA#29]
We are attempting to follow MicroLIA's documented examples to learn how to use the software with the goal of applying the software to a different dataset.  

Using the OGLE-II data and following the example steps leads to an issue when the CSV file is loaded.  
MicroLIA reports an attribute error saying that 'label' is missing.  

Does the training dataset format match the current code version?

**Comment by Professor-G** [comment]
I am not able to replicate this issue on a virtual environment. The latest version of MicroLIA on pip is 2.7, please ensure this is installed. This version generates the training set and by default saves the training data as a csv file on your local. That csv file should have a label column, which is what the Classifier class looks at when determining what the training labels should be.  

Please ensure the csv file is first loaded manually like so:

from pandas import read_csv

csv_file = read_csv('MicroLIA_Training_Set.csv')

Please note that when instantiating the Classifier, this loaded csv should be entered into the training_data argument:

from MicroLIA import ensemble_model

model = ensemble_model.Classifier(training_data=csv_file)

Is this where the error occurs?

**Comment by Professor-G** [comment]
I've identified the compatibility issues:

MicroLIA is currently compatible with Python 3.9 only, support for later versions is limited because pkg_resources (from setuptools) is no longer available. This module is used internally for accessing packaged data and is being replaced with importlib.resources. 

During the upgrade process I also identified that some upstream dependencies which rely on deprecated scipy functions that are no longer supported. I am in the process of replacing or modernizing these dependencies, and these changes will be included in an upcoming release that supports Python 3.12.

In the meantime, I have released MicroLIA version 2.7.2 to PyPI. This update includes explicit Python version constraints and a clearly defined list of pinned dependencies to ensure compatibility. If you are using Python 3.9 in a clean environment, installing MicroLIA via pip will now automatically install all required dependencies, including:

numpy
scikit-learn==1.1.1
scikit-optimize==0.9.0
astropy==5.0.4
scipy==1.7.3
peakutils==1.3.3
progress==1.6
matplotlib==3.5.1
optuna==3.1.0
boruta==0.3
BorutaShap==1.0.16
xgboost==1.6.1
scikit-plot==0.3.7
opencv-python==4.7.0.68
pandas==1.4.1
dill
gatspy==0.3
astroML==1.0.2.post1


Please also note that MicroLIA does not use Poetry as it is based on a setuptools build system. The updated setup.py file now explicitly handles dependency management. Installing MicroLIA with pip in a Python 3.9 environment should now complete without issues, please let me know of any other problems while I update the code to be compatible with modern Python versions.

Finally, due to backward compatibility with pickle, the test script test_classifier.py will fail when loading the test model, as it was saved on a different Python version than 3.9. While I confirmed all key functionality is working such as lightcurve feature extraction, as well as model saving/loading and optimization, unfortunately I found that the BorutaShap algorithm which is a powerful feature selection method has an unresolved bug, see: https://github.com/Ekeany/Boruta-Shap/issues/131

The fix requires changing the line
self.shap_values = self.shap_values.mean(0)
to
self.shap_values = self.shap_values.mean(1) 
in lines 721 and 745 of the BorutaShap module (.../site-packages/BorutaShap.py)

If you wish to avoid this bug you can skip the feature selection routine by setting the argument boruta_trials=0 when instantiating the Classifier class; although optimization is disabled by default, so this is only necessary if you set optimize=True. The upcoming version of MicroLIA will include this feature selection algorithm as a self-contained module to avoid such bugs in the future. I do appreciate your patience while I work through this.
